fix: update validate and evaluate to support target flag

CloudBeard opened the pull request:
## Description

Update the `validate` and `evaluate` action to check if `target` is empty or not. If `target` is empty processes as normal. If `target` exists will use the framework provided to run the `validate` and `evaluate` commands against that framework.

Changed the variable wording. The old target was the file `input-file`, with the addition of the `target` flag that could be confusing reading the code so refactored the variables to allow for `target` to represent the lula `target` flag and `oscal-file` to represent the `input-file`

removed the `-a` flags from the lula commands and replaced with `-o` to output the files correctly.

## Related Issue

Fixes #18 


## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging
Updated Documentation to include the new flag. 

Tested the bash script to ensure functions correctly.

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/lula-action/blob/main/CONTRIBUTING.md) followed

CloudBeard committed:
update validate and evaluate to support target flag

brandtkeller reviewed (changes requested):
I believe intent is correct - I'd like to get ahead of a few more configurations:

1. The output of validate should be configurable. This allows CI/Lula to merge assessments natively as part of validate output. 
2. I'd like to consider if we should split validate / evaluate or not. I don't hold a strong opinion at the moment but integration work we've done to-date may help influence them being done separately.

CloudBeard committed:
split out validate and evaluate

CloudBeard commented:
Split those into separate actions, updated docs, and added an options field to future proof it a bit.

brandtkeller commented on validate/action.yml:
```diff
@@ -1,41 +1,48 @@
+        file_dir="${HOME}/.lula-cache"
+        assessment-result="${{ inputs.assessment-result }}"
+        component-definition="${{ inputs.component-definition }}"
+        target="${{ inputs.target }}"
```
Calling out for discussion - is `target` something that the action needs to natively support if `options` exists? 

or rather _should_ it? I don't feel strongly either way but I am looking for how to reduce the amount of conditionals present in both actions.

brandtkeller reviewed (changes requested):
Appears valid but I'd either need to instrument a test repo or run locally. As such I am going to create an issue to create a workflow for testing the actions that can run in these pipelines. 

I tossed a few comments out for discussion - push back as you see fit but i'd like to reduce complexity as much as possible.

CloudBeard committed:
initial testing for workflows

CloudBeard committed:
still wip

CloudBeard committed:
location issue

CloudBeard committed:
test dir

CloudBeard committed:
wrong spot

CloudBeard committed:
still testing

CloudBeard committed:
using use wrong

CloudBeard committed:
cant spell

CloudBeard committed:
progress

CloudBeard committed:
testing - to _

CloudBeard committed:
missing an conditional closure

CloudBeard committed:
fix validate test

CloudBeard committed:
make lint follow similiar - to _

CloudBeard committed:
test remove target for validate

CloudBeard committed:
closing fi

CloudBeard committed:
passing options in as a string

CloudBeard committed:
testing with blank options

CloudBeard committed:
added more use cases and testing evaluate

CloudBeard committed:
remove extra if

CloudBeard committed:
cleanu

CloudBeard commented:
Removed target and added multiple tests in the workflow to cover a few possible options.

Should also close https://github.com/defenseunicorns/lula-action/issues/23

brandtkeller reviewed (approved):
Appreciate the iterations that lead to cleaning these up.
